packages/engine/src/interaction/components/InteractableComponent.ts

```diff
@@ -145,17 +180,33 @@ export const InteractableComponent = defineComponent({
     const entity = useEntityContext()
     const interactable = useComponent(entity, InteractableComponent)
     const input = useOptionalComponent(entity, InputComponent)
+    const isEditing = useMutableState(EngineState).isEditing
+
+    useEffect(() => {
+      setComponent(entity, DistanceFromCameraComponent)
+      setComponent(entity, DistanceFromLocalClientComponent)
+
+      addInteractableUI(entity)
+    }, [])
+
+    useEffect(() => {
```

Comment:
this should be refactored to use cleanup hooks
```ts
 useEffect(() => {
      if (isEditing.value) return
      addInteractableUI(entity)
      return () => {
        removeInteractableUI(entity)
      }
    }, [isEditing.value])    
```
this will also make the `onRemove` cleanup unnecessary